proxy/src/test/java/com/velocitypowered/proxy/scheduler/VelocitySchedulerTest.java

```diff
@@ -65,4 +67,36 @@ void repeatTaskWorks() throws Exception {
     task.cancel();
   }
 
+  @Test
+  void obtainTasksFromPlugin() throws Exception {
+    VelocityScheduler scheduler = new VelocityScheduler(new FakePluginManager());
+    AtomicInteger i = new AtomicInteger(0);
+
+    scheduler.buildTask(FakePluginManager.PLUGIN_A, task -> {
+      if (i.getAndIncrement() >= 1) {
+        task.cancel();
+      } else {
+        scheduler.buildTask(FakePluginManager.PLUGIN_A, () ->
+          assertEquals(scheduler.tasksByPlugin(FakePluginManager.PLUGIN_A).size(), 1)
+        ).schedule();
+      }
+    }).delay(50, TimeUnit.MILLISECONDS)
+      .repeat(Duration.ofMillis(5))
+      .schedule();
+  }
+
+  @Test
+  void testConsumerCancel() throws Exception {
+    VelocityScheduler scheduler = new VelocityScheduler(new FakePluginManager());
+
+    scheduler.buildTask(FakePluginManager.PLUGIN_B, actualTask -> {
```

Comment:
I'd also have to have a condition to check if `actualTask` is the same value as what `TaskBuilder#scheduler()` will return.